lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -46,34 +47,101 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
                 filter_func=possibly_scored,
                 start_node=self.location,
         ):
-            descendant = course_structure[descendant_key]
+            self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
-            if not getattr(descendant, 'has_score', False):
-                continue
+        self.all_total, self.graded_total = graders.aggregate_scores(self.scores, self.display_name, self.location)
 
+    def save(self, student, subsection, course):
+        """
+        Persist the SubsectionGrade.
+        """
+        visible_blocks = [
+            BlockRecord(unicode(location), weight, score.possible)
```

Comment:
There are multiple ways to handle this.

Typically, any in-memory representation of an OpaqueKey should be an OpaqueKey object - with serialization being only at the far edge.  

@jcdyer  How did you expect callers to use the `BlockRecord` interface?
I can see @efischer19's point in preparing for serialization.  However, then the `locator` field in the `BlockRecord` should be renamed to `locator_string` or some such..  The interface to the data model should then be clear that all callers should serialize the data beforehand.

Alternatively, we can change `BlockRecord` to be a class and have it implement its own serialize method.